akka-remote/src/main/scala/akka/remote/artery/TestStage.scala

```diff
@@ -173,11 +172,25 @@ private[remote] class InboundTestStage(inboundContext: InboundContext, state: Sh
                   push(out, env)
               case _ =>
                 // unknown, handshake not completed
-                if (state.anyBlackholePresent())
-                  log.debug(
-                    "inbound message [{}] before handshake completed, cannot check if remote is blackholed, letting through",
-                    Logging.messageClassName(env.message))
-                push(out, env)
+                if (state.anyBlackholePresent()) {
+                  env.message match {
+                    case _: HandshakeReq =>
+                      log.debug(
+                        "inbound message [{}] before handshake completed, cannot check if remote is blackholed, letting through",
+                        Logging.messageClassName(env.message))
+                      push(out, env)  // let it through
+
+                    case anyOther =>
+                      log.debug(
+                        "dropping inbound message [{}] with UID [{}] because of blackhole",
+                        Logging.messageClassName(anyOther),
+                        env.originUid
+                      )
+                      pull(in) // drop message
+                  }
+                } else {
+                  push(out, env)
+                }
```

Comment:
what should we do if we don't have the association, but we know it has some blackholes? 

The original idea was to let `HandshakeReq` pass and drop the rest. At least that's what I understood from the description in https://github.com/akka/akka/issues/29963. So implemented that way. 

However, @johanandren raised the idea to simply drop all messages if we reach that situation. 

Anyway, happy to change it if we think it's better.